### DISCUSSION TITLE
Binding to "parent" data context in ItemsRepeater

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#11339] [Q&A] [answered]
I want to bind some elements of a data item template NOT to items data context but to the "parent" kind of data context. In below sample specifically to a command in the general data context using command parameter to identify on which item it is initiated. This seems often the better solution compared to define that within the item data class. 

I wonder whether the binding syntax I used 

```csharp
Command="{Binding ElementName=itemsParent, Path=DataContext.SetVisibilityDisabledCommand}"
```
is the best way to do it!?

```csharp
          <ItemsRepeater Items="{Binding MyItems}"
                         Name="itemsParent">
            <ItemsRepeater.Layout>
              <StackLayout Spacing="8"
                  Orientation="Vertical" />
            </ItemsRepeater.Layout>
            <ItemsRepeater.ItemTemplate>
              <DataTemplate>
                <bui:StretchPanel Orientation="Horizontal"
                                  ItemSpacing="4">
                  <TextBlock Classes="menuItem"
                             Padding="0 4 0 0 "
                             HorizontalAlignment="Left"
                             Text="{Binding DisplayName, Converter={StaticResource I18nMand}}"/>
                  <Grid>
                    <ucp:DefaultRadioButton Classes="menuItem"
                                            IsVisible="{Binding Mode, Mode=OneWay, Converter={StaticResource xrayVisualizationModeBoolConverter}, ConverterParameter=Undefined}"
                                            Command="{Binding ElementName=itemsParent, Path=DataContext.SetVisibilityEnabledCommand}"
                                            CommandParameter="{Binding VisualizationState}"/>
                    <ucp:DefaultRadioButton Classes="menuItem"
                                            IsVisible="{Binding Mode, Mode=OneWay, Converter={StaticResource xrayVisualizationModeBoolConverter}, ConverterParameter=Enabled}"
                                            Command="{Binding ElementName=itemsParent, Path=DataContext.SetVisibilityDisabledCommand}"
                                            CommandParameter="{Binding VisualizationState}"/>
                    <ucp:DefaultRadioButton Classes="menuItem"
                                            IsVisible="{Binding Mode, Mode=OneWay, Converter={StaticResource xrayVisualizationModeBoolConverter}, ConverterParameter=Disabled}"
                                            Command="{Binding ElementName=itemsParent, Path=DataContext.SetVisibilityUndefinedCommand}"
                                            CommandParameter="{Binding VisualizationState}"/>
                  </Grid>
                </bui:StretchPanel>
              </DataTemplate>
            </ItemsRepeater.ItemTemplate>
          </ItemsRepeater>
```

**Comment by timunie** [comment, marked as answer]
`{Binding $parent[ItemsRepeater].((vm:MainViewModel)DataContext).MyCommand}` 
or
`{Binding #ItemParent.((vm:MainViewModel)DataContext).MyCommand}`

both should work imo

**Reply by llfab** [reply]
Thanks. This here works an seems most simple: 
```Command="{Binding $parent[ItemsRepeater].DataContext.SetVisibilityEnabledCommand}"```

(the manual cast seems not necessary)

**Reply by timunie** [reply]
It is if you enable compiled bindigs. Which I highly advice btw

**Reply by llfab** [reply]
Aren't compiled bindings enabled by default now?  I thought I read something the other day. At least I have never set them explicitly. 

If I were to set them, would I set them in each UserControl or just the MainWindow?

**Reply by timunie** [reply]
in a new project, yes. But then you need casting afair

**Reply by llfab** [reply]
Ok. I asked too many questions at once:
- If not set anywhere, a 11.0.0-preview6+ app has Compiled Bindings enabled by default, right? At least in my project which I kept up to date with new previews, the non-casted binding works
- If one wants to use non-compiled bindings that where to be set in a) each UserControl or b) just MainWindow?

**Reply by llfab** [reply]
Ok. I understood wrong. The data type explicate conversion, you need for Compiled Bindings. That is why it just works on my end. Sorry for being slow.
